### ISSUE TITLE
Whisper transcription app big performance regression

### ISSUE THREAD

**Issue by thundergolfer** [modal-labs/modal-examples#712]
https://modal-com.slack.com/archives/C069RAH7X4M/p1713624663717089

<kbd>
<img width="790" alt="image" src="https://github.com/modal-labs/modal-examples/assets/12058921/ab20a395-cb59-45aa-9b7a-a9290a3ece61">
</kbd>

**Comment by thundergolfer** [comment]
A one hour podcast used to take ~1 minute, so big drop in performance.

**Comment by thundergolfer** [comment]
I think first thing to do is to replace the use of NFS

**Comment by ahxxm** [comment]
would be great if the official example uses WhisperX, it can transcribe one hour podcast in 1 minute using only 1 container, instead of `spins up 100-300 containers for a single transcription`

**Comment by ahxxm** [comment]
made a poc repo here https://github.com/ahxxm/serverless-audio-transcriber

**Comment by ahxxm** [comment]
I've been using my own dogfood for a while, this is how it looks like with A10G

The audio files I sent range from 30 minutes to 70 minutes, and I scheduled a 5 minute interval

![{06617E04-3793-49C8-968B-86164686CABE}](https://github.com/user-attachments/assets/110531de-9c4d-4a73-a49b-a2cbebf0e784)

This translates to around $0.01~$0.02 per hour of transcription, wonder how does the current official approch look like? Before and after regression, will it be cheaper and faster?

**Comment by thundergolfer** [comment]
Thanks @ahxxm this is awesome, especially the bit we're you've listed [benchmarks](https://github.com/ahxxm/serverless-audio-transcriber?tab=readme-ov-file#benchmarks-and-caveats). It looks like Runpod is the cheapest? I'd bet that the 
RTX A4500 $/hr rate is cheaper than our A10G.

**Comment by ahxxm** [comment]
Yeah A4500 is a new, performant(for Whisper) and cheap(Modal T4 price) one, would be great if Modal also supports that. 

I think it's comparable but not a completely fair comparison, Runpod has FlashBoot that loads weights within 1 second instead of 20+ seconds, so I added a bit more CPU and memory to Modal codes

**Comment by gongouveia** [comment]
@thundergolfer @ahxxm hello, thank your for this very informative thread. I would like to ask how could I use my own model using modal, how can i send my file to the container image?

**Comment by ahxxm** [comment]
@gongouveia I doubt this is relevant to the issue, but are you asking about how to send custom model/weights [into Docker image](https://github.com/ahxxm/serverless-audio-transcriber/blob/master/Dockerfile#L13), or send the file you want to transcribe(see all examples in this repo or mine then)?

**Comment by charlesfrye** [comment]
Closing this as stale.

**Comment by ahxxm** [comment]
hey just because it's stale doesn't mean the problem is solved, I'd still like to see official examples use resources more efficiently and , since they are official examples, encourage people to do so.

Instead of 

> spins up 100-300 containers for a single transcription

Why not transcribe using single graphics card in 1 minute?

**Comment by charlesfrye** [comment]
You're welcome to submit a PR!

**Comment by ahxxm** [comment]
no thanks, it's surely out of my scope
